Formula/j/janet.rb

```diff
@@ -17,30 +18,57 @@ class Janet < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "31581c3ce8ea833fe2a60d42ce0879093be588c37d4336910dc59f891aa40eba"
   end
 
-  depends_on "meson" => :build
-  depends_on "ninja" => :build
-
   resource "jpm" do
     url "https://github.com/janet-lang/jpm/archive/refs/tags/v1.1.0.tar.gz"
     sha256 "337c40d9b8c087b920202287b375c2962447218e8e127ce3a5a12e6e47ac6f16"
   end
 
+  def syspath
+    HOMEBREW_PREFIX/"lib/janet"
+  end
+
   def install
-    system "meson", "setup", "build", *std_meson_args
-    cd "build" do
-      system "ninja"
-      system "ninja", "install"
-    end
+    # Replace lines in the Makefile that attempt to create the `syspath`
+    # directory (which is a directory outside the sandbox).
+    inreplace "Makefile", /^.*?\bmkdir\b.*?\$\(JANET_PATH\).*?$/, "# Line removed by Homebrew formula"
```

Comment:
maybe just comment out?
```suggestion
    inreplace "Makefile", /^.*?\bmkdir\b.*?\$\(JANET_PATH\).*?$/, "#"
```

Reply:
Not sure this one really matters, but if necessary it should be done in a follow-up.

Reply:
@chenrui333 Do you mean just match against `/^.*?\bmkdir\b.*?\S\(JANET_PATH\)/` and not bother with matching to the end of the line? Or is there a function I can use to insert a character into the first column of the line that has matched?

Reply:
yeah, I was just thinking about this comment `# Line removed by Homebrew formula` (it does not be homebrew specific though? 🤔 )

Reply:
Well, it's necessary to comment it out for Homebrew because the effect of this line in the Makefile is that it tries to create a directory outside of the keg (given the prefix I'm using) and the Homebrew sandbox won't allow that.

Reply:
So it seemed helpful to note that this was specifically because of Homebrew's needs and not something broken in the Makefile itself.